WordPress/Classes/Utility/CoreDataManager.swift

```diff
@@ -0,0 +1,391 @@
+
+/// Handles the core data stack for the whole app
+class CoreDataManager: CoreDataStack {
+
+    static let shared = CoreDataManager()
+
+    private init() {
+        observe()
+    }
+
+    private lazy var writerContext: NSManagedObjectContext = {
+        let context = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        context.persistentStoreCoordinator = self.persistentStoreCoordinator
+        return context
+    }()
+
+    var mainContext: NSManagedObjectContext {
+        persistentContainer.viewContext
+    }
+
+    var persistentStoreCoordinator: NSPersistentStoreCoordinator {
+        persistentContainer.persistentStoreCoordinator
+    }
+
+    var managedObjectModel: NSManagedObjectModel {
+        guard let mom = NSManagedObjectModel(contentsOf: modelURL) else {
+            fatalError("Could not load model")
+        }
+        return mom
+    }
+
+    lazy var persistentContainer: NSPersistentContainer = {
+        migrateDataModelIfNecessary()
+
+        let container = NSPersistentContainer(name: Constants.name, managedObjectModel: managedObjectModel)
+        container.persistentStoreDescriptions = [storeDescription]
+
+        container.loadPersistentStores { [weak self] (storeDescription, error) in
+            guard let `self` = self, let error = error else {
+                return
+            }
+            DDLogError("⛔️ [CoreDataManager] loadPersistentStore failed. Attempting to recover... \(error)")
```

Comment:
One thing that we need to add here and test is the usage of [`SentryStartupEvent`](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/Logging/SentryStartupEvent.swift), the same way as we do in the `ContextManager.m`.